manager.c

```diff
@@ -182,12 +181,15 @@ static int manager_send_sms (struct mansession* s, const struct message* m)
 		return 0;
 	}
 
-	char * unescaped_msg = unescape_newlines(message);
+	size_t msg_size = strlen(message) + 1;
+	char* unescaped_msg = ast_malloc(msg_size);
+	ast_copy_string(unescaped_msg, message, msg_size);
```

Comment:
ast_malloc + ast_copy_string = ast_strdup

Further, every malloc/strdup needs a check for NULL, for the rare cases when you're out of memory.